src/opm/parser/eclipse/EclipseState/Aquancon.cpp

```diff
@@ -18,107 +18,98 @@
  */
 #include <opm/parser/eclipse/EclipseState/Grid/FaceDir.hpp>
 #include <opm/parser/eclipse/EclipseState/Aquancon.hpp>
+
+#include <unordered_map>
 #include <utility>
 #include <algorithm>
 #include <iterator>
 #include <iostream>
 
 namespace Opm {
+
     namespace{
-        struct AquanconRecord{
-            std::vector<size_t> global_index_per_record;
 
-            // Variables constants
-            std::vector<std::shared_ptr<double>>  influx_coeff_per_record;  //Aquifer influx coefficient
-            std::vector<double>  influx_mult_per_record;   //Aquifer influx coefficient Multiplier
-            // Cell face to connect aquifer to
-            std::vector<int> face_per_record;
-            std::vector<int> record_index_per_record;
-        };
+        void add_cell(std::unordered_map<std::size_t, Aquancon::AquancCell>& work, Aquancon::AquancCell cell) {
+            auto cell_iter = work.find(cell.global_index);
+            if (cell_iter == work.end())
+                work.insert( std::make_pair(cell.global_index, std::move(cell)));
+            else {
+                /*
+                 */
+                auto& prev_cell = cell_iter->second;
+                if (prev_cell.aquiferID == cell.aquiferID) {
```

Comment:
I was a bit uncertain when trying to understand the existing implementation/manual - but my interpretation is that the second time a cell is attached to an aquifer the connection is ignored if it is a new aquifer - i.e. considering pairs of cells and aquifer as `(cell, aquifer)` then in the following situation:
```
(cell1, aquifer1)
...
(cell1, aquifer2)
```
the second coupling `(cell1, aquifer2)` will be silently ignored. I have added a test for it and also some documentation.

As I said I found it difficult to interpret this - both from existing code and manual; If you think I have got it wrong that might very well be the case.

Reply:
> the second coupling (cell1, aquifer2) will be silently ignored. 

When two connections are specified in the grid cell for the same aquifer, we add the influx coefficients together.  Basically, it means, for one aquifer, we can only have one connection in the same grid blocks.  However, the previous input is not just ignored. (the document is not a little conflicting for this part)

I guess two aquifers might be able to have the connections in this same grid cell. But we do not have a case to verify it, and not sure how often it will happen. Have a `throw` in the `else` branch can postpone the issue to be handled later. Otherwise, we just lose some connections without noticing it. I suggest we add an `else` branch here. 

The old implementation is very hard to follow when coming to details.

Reply:
> When two connections are specified in the grid cell for the same aquifer, we add the influx coefficients together.

Yes - that is already implemented in the current code.

Reply:
> Have a throw in the else branch can postpone the issue to be handled later.

We could do that - but my interpretation of the manual is that in this situation the second connection should be ignored. If you thank a throw is better I will add that.

Reply:
>  If you thank a throw is better I will add that.

I think so. Please do it. We might come back to deal with it when we see the situation.

Reply:
I do not have more comment.

Reply:
I have added the `throw` behavior when one cell is connected to two different aquifers.